.github/workflows/build.yml

```diff
@@ -62,6 +62,14 @@ jobs:
       - name: Build PKI packages
         run: ./build.sh --with-timestamp --work-dir=build rpm
 
+      - name: Upload PKI packages
+        uses: actions/upload-artifact@v2
+        with:
+          name: pki-build-${{ matrix.os }}
+          path: |
+            build/RPMS/
+            build/SRPMS/
+
```

Comment:
We plan to access the rpms after the workflow are completed? If not the cache should be preferred. Among the benefit of the cache it should not stop if you reach the size limit but delete the oldest entries.

Reply:
Yes, the RPMs and SRPM are used in the new RPMInspect test: https://github.com/dogtagpki/pki/blob/master/.github/workflows/tools-tests.yml#L1423-L1448

Reply:
There was a suggestion to include the RPMInspect job as part of the `build.yml` workflow, so this would be unnecessary, but I disagreed with it as it would extend the length of the workflow by ~7 minutes - preventing all the subsequent jobs from starting. It is probably better to put that test in `code-analysis.yml` instead of `tools-tests.yml`, I will probably do so at some point.

Reply:
Actually, there is not `build.yml` workflow but it is part of other workflows. Therefore, each workflow including `build.yml` will generate its copy of the artifact. I have a look in you repository and the upload is ~20MB and this for each workflow so every PR and merge is like 300 MB. I think we will end the storage available quickly.

The `RPMInspect` now uses the artifact of its workflow so only one is publishing.

I think other approach should be considered. Maybe build in a separate workflow and all the other get the artifact/cache entries, or modify in a single workflow.

Reply:
Yeah, `code-analysis.yml` probably would be a good place for RPMInspect.

So IIUC the build will be executed multiple times by different tests, but the artifacts will be uploaded with the same name (i.e. `pki-build-${{ matrix.os }}`), would there be any concern about conflicts/corruptions? Or should we include `key_container` in the name too just like the cache?

Reply:
> would there be any concern about conflicts/corruptions?

AFAIK this is a problem with the cache which is shared among the workflows. The artifact should be per workflow.